### ISSUE TITLE
By using sample weight, why multiply a constant changed the model ?

### ISSUE THREAD

**Issue by dtss316** [microsoft/LightGBM#6338]
## Description
Hi,

I am using LightGBM to train a model and setting reg_lambda and reg_alpha both to 0, while choosing regression as the objective function. Then, based on my own logic, I generated sample weights. I have tested two scenarios: one using sample_weight and the other using sample_weight multiplied by 0.9999. However, there is a noticeable difference in the training results between these two cases. According to my understanding, if reg_lambda and reg_alpha are set to 0, multiplying sample_weight by a constant should not change the training results.

Could someone help me with this? Thanks.

## Environment info

LightGBM version or commit hash:

Command(s) you used to install LightGBM

```shell

```

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

1. what version are you using?
2. how did you install it (exact commands)?
3. what programming language are you using?
4. can you provide a minimal, reproducible example showing exactly how you're using LightGBM?

**Comment by shiyu1994** [comment]
It is possible that multiplying 0.99999 causes a difference in the gradient accumulation process. Though theoretically this should have no effect on the predicted values, but practically floating-point numbers have a finite precision. Multiplying a series of numbers by 0.99999, summing them up together, and dividing them by 0.99999 will not necessarily returns the same results as the original summation of these numbers.

**Comment by shiyu1994** [comment]
But again, the difference should be minor. If you encounter large difference in result, then providing more details or a reproducible example could be very helpful for us to identify the problem.
